packages/remix-dev/vite/plugin.ts

```diff
@@ -822,6 +822,16 @@ export const remixVitePlugin: RemixVitePlugin = (options = {}) => {
     },
     {
       name: "remix-hmr-updates",
+      async configureServer(server) {
+        // trigger full reload when adding a new route file
+        server.watcher.addListener("add", async (file: string) => {
+          cachedPluginConfig = await resolvePluginConfig();
+          let route = getRoute(cachedPluginConfig, file);
+          if (route) {
+            server.ws.send({ type: "full-reload" });
+          }
+        });
+      },
```

Comment:
I'm not sure proper attribution is mandatory for this small code, but I personally look up rakkasjs quite often for vite techniques and this specific idea is based on this code:

https://github.com/rakkasjs/rakkasjs/blob/3030cae9ec4a232051e99956130ecd3d74786ebf/packages/rakkasjs/src/features/api-routes/vite-plugin.ts#L132-L153